### DISCUSSION TITLE
How to import different global css for certain stories?

### DISCUSSION THREAD

**Discussion by antonbehaeghe** [storybookjs/storybook#17062] [Help]
Hi,

I'm working on a monorepo and we have setup one storybook package that loads all stories from all other packages in the repo. For some of these, I would like to import another global css file. For this I thought I would need a decorator, that then imports said css file, however webpack just adds this to all other css imports it can find, so the css is applied to all stories.

How can I solve this? Is there any way to 'unload' the css when switching to a different story? Can I use the preview-head.html for this and dynamically update it using decorators?

Or do I have to run separate storybooks / use env vars to load different config when I want to work on these stories?

Any help would be greatly appreciated! 🙏

**Comment by antonbehaeghe** [comment]
I've got it working by using a decorator to import the css as a string, and then add an inline style tag with the css.

Using an inline webpack loader here to bypass other webpack config.

```
import css from '!css-loader!./src/main.css';

export function Layout({ children }: { children: React.ReactElement }) {
  return (
    <div >
      <style>{css.toString()}</style>
      {children}
    </div>
  );
}

**Reply by joanrodriguez** [reply]
Thanks this solved it for me!

**Reply by sanbornhilland** [reply]
This also solved it for me. I have some stories that read CSS variables so I had to modify this to force the component to re-render once the `style` tag is parsed. It's not a pretty hack but it works:

```tsx
import { ComponentType, ReactNode, useState } from "react"
// @ts-expect-error
import css from "!css-loader!../index.css"

export function GlobalStyles({ children }: { children: ReactNode }) {
  return (
    <>
      <style>{css.toString()}</style>
      {children}
    </>
  )
}

export function GlobalStylesDecorator(Story: ComponentType) {
  // Because we are dynamically inserting the style tag and then immediately
  // rendering the Story, stories that are using JS to read from the contents
  // of the style tag will find the tag is empty. This is a small hack to force
  // a re-render of the component until after the style tag is parsed.
  const [, forceRender] = useState(0)

  setTimeout(() => {
    forceRender(1)
  }, 1)

  return (
    <GlobalStyles>
      <Story />
    </GlobalStyles>
  )
}
```
